File: php/slim-doctrine/app/src/com/sprint/sms/api/domain/Access.php
<?php
namespace App\com\sprint\sms\api\domain;

use Doctrine\ORM\Mapping as ORM;

use App\com\sprint\sms\api\base\domain\BaseEntry;
use App\com\sprint\sms\api\support\FormatSupport;

use \App\com\sprint\sms\api\domain\User;

/**
 * @ORM\Entity
 * @ORM\Table(name = "t_access")
 */
class Access extends BaseEntry implements FormatSupport
{

	/**
     * @ORM\Column(name="ID_", type = "string", nullable = false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy = "UUID")
     */
	private $id;
	
	/**
	 * @ORM\ManyToOne(targetEntity = "User", fetch = "EAGER")
	 * @ORM\JoinColumn(name = "f_user", referencedColumnName = "ID_", unique = true, nullable = false, onDelete="CASCADE")
	 */ 
    private $user;
	
	/**
     * @ORM\Column(name = "agent", type = "string", length = 1024, nullable = false)
     */
    private $agent;
	
	/**
     * @ORM\Column(name = "expired", type = "bigint", nullable = false, options = {"unsigned":true, "default":0})
     */
	private $expired;
	
	public function setId($id) {
		$this->id = $id;
	}
	
	public function getId() {
		return $this->id;
	}    
	
	public function setUser(User $user)
    {
        $this->user = $user;
    }
	
    public function getUser()
    {
        return $this->user;
    }
	
	public function setAgent($agent)
    {
        $this->agent = $agent;
    }

    public function getAgent()
    {
        return $this->agent;
    }	
	
	public function setExpired($expired)
    {
        $this->expired = $expired;
    }   
	
	public function getExpired()
    {
        return $this->expired;
    }   
	
	public function hasExpired() {
		$now = round(microtime(true) * 1000);
		return $now > $this->expired;
	}
	
	public function toFormatObject() {
		$o = parent::toFormatObject();
		$o["id"] = $this->id;
		$o["user"] = isset($this->user) ? $this->user->toFormatObject() : null;
		$o["agent"] = $this->agent;	
		return $o;
	}
}

File: java/spring-hibernate/src/main/java/com/sprint/sms/api/domain/Access.java
package com.sprint.sms.api.domain;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import com.ideahut.common2.annotation.IdhFormatter;
import com.ideahut.shared2.domain.base.StringIdEntryDomain;

/*
 * ID yang akan digunakan sebagai AccessKey
 * User unik -> Agent diperoleh dari request header 'User-Agent'
 * Validasi dari Access Key pada saat request adalah Agent
 */

@Entity
@Table(name = "t_access")
@IdhFormatter
public class Access extends StringIdEntryDomain {

	/**
	 * 
	 */
	private static final long serialVersionUID = 231481428978413107L;

	private User user;
	
	private String agent;
	
	private Long expired;

	@ManyToOne(targetEntity = User.class, fetch = FetchType.EAGER)
	@OnDelete(action = OnDeleteAction.CASCADE)
	@JoinColumn(name = "f_user", nullable = false, unique = true)
	@IdhFormatter
	public User getUser() {
		return user;
	}
	
	public void setUser(User user) {
		this.user = user;
	}

	@Column(name = "agent", length = 1024, nullable = false)
	@IdhFormatter
	public String getAgent() {
		return agent;
	}

	public void setAgent(String agent) {
		this.agent = agent;
	}

	@Column(name = "expired", nullable = false)
	@IdhFormatter
	public Long getExpired() {
		return expired;
	}

	public void setExpired(Long expired) {
		this.expired = expired;
	}
	
	public boolean hasExpired() {
		if (expired == null) {
			return true;
		}
		return System.currentTimeMillis() > expired;
	}
	
}